tuning/utils/merge_model_utils.py

```diff
@@ -102,3 +102,77 @@ def fetch_base_model_from_checkpoint(checkpoint_model: str) -> str:
             "Base model adapter config exists, but has no base_model_name_or_path!"
         )
     return adapter_dict["base_model_name_or_path"]
+
+def _copy_files_to_directory(src: str , dest: str, exclude_files:list[str] = []):
+    import shutil
+    src_files = os.listdir(src)
+    for file_name in src_files:
+        if file_name in exclude_files:
+            continue
+        full_file_name = os.path.join(src, file_name)
+        if os.path.isfile(full_file_name):
+            shutil.copy(full_file_name, dest)
+
+def post_process_vLLM_adapters_new_tokens(path_to_checkpoint: str, modified_checkpoint_path:str = None):
+
+    from safetensors import safe_open
+    from safetensors.torch import save_file
+
+    # if not set, original checkpoint will be modified
+    if not modified_checkpoint_path:
+        modified_checkpoint_path = path_to_checkpoint
+
+    # Get all values of new token indexes
+    sorted_token_indexes = []
+    if os.path.isfile(os.path.join(path_to_checkpoint,"added_tokens.json")):
+        with open(os.path.join(path_to_checkpoint,"added_tokens.json"), "r") as fp:
+            added_tokens = json.load(fp)
+            sorted_token_indexes = sorted(added_tokens.values())
+
+    with safe_open(os.path.join(path_to_checkpoint, "adapter_model.safetensors"), framework="pt") as f:
+        new_embeddings = {}
+        adapters = {}
+        embeddings_weights_in_adapters = False
+        # Quickly check if post-processing is needed by checking adapters file for weights
+        for k in f.keys():
+            if 'lm_head.weight' in k or 'embed_tokens.weight' in k:
+                embeddings_weights_in_adapters = True
+                if len(sorted_token_indexes) >=1:
+                    raise NotImplementedError("Seems like embeddings are resized without adding new tokens. \
+                                               Cannot be post-processed to load on vLLM. Try setting \
+                                               parameter `embedding_size_multiple_of` to 1" )
+
+        # Post-processing is needed to copy out new vectors
+        if embeddings_weights_in_adapters:
+            for k in f.keys():
+                if 'lm_head.weight' in k:
+                    lm_head = f.get_tensor(k)
+                    # pull out tensor values of new tokens
+                    if len(sorted_token_indexes)==1:
+                        new_output_embeddings = lm_head[sorted_token_indexes[0]:sorted_token_indexes[0]+1]
+                    elif len(sorted_token_indexes)>1:
+                        new_output_embeddings = lm_head[sorted_token_indexes[0]:sorted_token_indexes[-1]]
```

Comment:
This should work right? 🤔 
```suggestion
                    new_output_embeddings = lm_head[sorted_token_indexes[0]:sorted_token_indexes[-1]+1]
```